Utilities/DataLoading/EdFi.LoadTools/ApiClient/ResourceToInterchangeMapProvider.cs

```diff
@@ -2,69 +2,213 @@
+            new Dictionary<string, List<FileTypeInfo>>();
+        private readonly Lazy<Dictionary<string, List<FileContext>>> _resourceNameByFileContext;
+        private readonly XmlSchemaSet _xmlSchemaSet;
+        private readonly IXsdConfiguration _xsdConfiguration;
 
-        public ResourceToInterchangeMapProvider(XmlSchemaSet xmlSchemaSet, IResourceStreamFactory resourceStreamFactory)
+        public ResourceToFileTypeInfoProvider(XmlSchemaSet xmlSchemaSet, IDataConfiguration dataConfiguration,
+            IXsdConfiguration xsdConfiguration)
         {
             _xmlSchemaSet = xmlSchemaSet;
-            _resourceStreamFactory = resourceStreamFactory;
+            _dataConfiguration = dataConfiguration;
+            _xsdConfiguration = xsdConfiguration;
+
+            _resourceNameByFileContext =
+                new Lazy<Dictionary<string, List<FileContext>>>(GetFileTypeByFileContext);
         }
 
-        public Dictionary<string, List<InterchangeInfo>> GetResourceToInterchangeMap()
+        public Dictionary<string, List<FileTypeInfo>> GetResourceByFileTypeInfo()
         {
-            var interchangeToInterchangeFilesMap = _resourceStreamFactory.GetInterchangeToInterchangeFilesMap();
-
             foreach (XmlSchema schema in _xmlSchemaSet.Schemas())
             {
                 foreach (XmlSchemaElement element in schema.Elements.Values)
                 {
-                    var interchangeName = element.Name;
+                    var fileType = element.Name;
 
                     var resources = element.ElementSchemaType.GetUnorderedElementNames().ToList();
 
-                    resources.ForEach(r =>
+                    resources.ForEach(
+                        r =>
+                        {
+                            var resource = r.ToLower();
+
+                            if (!_resourceByFileTypeInfo.TryGetValue(resource, out List<FileTypeInfo> fileTypeInfos))
+                            {
+                                fileTypeInfos = _resourceByFileTypeInfo[resource] = new List<FileTypeInfo>();
+                            }
+
+                            if (!_resourceNameByFileContext.Value.TryGetValue(fileType, out List<FileContext> fileContexts))
+                            {
+                                fileContexts = new List<FileContext>();
+                            }
+
+                            fileTypeInfos.Add(
+                                new FileTypeInfo
+                                {
+                                    FileType = fileType,
+                                    FileContexts = fileContexts
+                                });
+                        });
+                }
+            }
+
+            return _resourceByFileTypeInfo;
+        }
+
+        private Dictionary<string, List<FileContext>> GetFileTypeByFileContext()
+        {
+            var fileTypeByFileContexts = new Dictionary<string, List<FileContext>>();
```

Comment:
Dictionary name seems strange here too, shouldn't the names be flipped?

Reply:
done